lm_eval/tasks/wino_x.py

```diff
@@ -0,0 +1,141 @@
+
+See: https://aclanthology.org/2021.emnlp-main.670.pdf
+
+"""
+import numpy as np
+from lm_eval.base import rf, Task
+from lm_eval.metrics import mean
+
+
+_CITATION = """
+@inproceedings{emelin-sennrich-2021-wino,
+    title = "Wino-{X}: Multilingual {W}inograd Schemas for Commonsense Reasoning and Coreference Resolution",
+    author = "Emelin, Denis  and
+      Sennrich, Rico",
+    booktitle = "Proceedings of the 2021 Conference on Empirical Methods in Natural Language Processing",
+    month = nov,
+    year = "2021",
+    address = "Online and Punta Cana, Dominican Republic",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/2021.emnlp-main.670",
+    doi = "10.18653/v1/2021.emnlp-main.670",
+    pages = "8517--8532",
+    abstract = "Winograd schemas are a well-established tool for evaluating coreference resolution (CoR) and commonsense reasoning (CSR) capabilities of computational models. So far, schemas remained largely confined to English, limiting their utility in multilingual settings. This work presents Wino-X, a parallel dataset of German, French, and Russian schemas, aligned with their English counterparts. We use this resource to investigate whether neural machine translation (NMT) models can perform CoR that requires commonsense knowledge and whether multilingual language models (MLLMs) are capable of CSR across multiple languages. Our findings show Wino-X to be exceptionally challenging for NMT systems that are prone to undesirable biases and unable to detect disambiguating information. We quantify biases using established statistical methods and define ways to address both of these issues. We furthermore present evidence of active cross-lingual knowledge transfer in MLLMs, whereby fine-tuning models on English schemas yields CSR improvements in other languages.",
+}
+"""
+
+
+class WinograndeXDe(Task):
+    VERSION = 0
+    DATASET_PATH = "demelin/wino_x"
+    DATASET_NAME = "lm_en_de"
+
+    answer_to_num = {1: 0, 2: 1}
+
+    def has_training_docs(self):
+        return False
+
+    def has_validation_docs(self):
+        return False
+
+    def has_test_docs(self):
+        return True
+
+    def training_docs(self):
+        if self._training_docs is None:
+            self._training_docs = list(self.dataset["train"])
+        return self._training_docs
+
+    def test_docs(self):
+        return self.dataset["test"]
+
+    def doc_to_text(self, doc):
+        return self.partial_context(doc, doc["option" + str(doc["answer"]) + "_de"])
+
+    def should_decontaminate(self):
+        return True
+
+    def doc_to_decontamination_query(self, doc):
+        return doc["context_de"]
+
+    @classmethod
+    def partial_context(cls, doc, option):
+        # Substitute the pronoun in the sentence with the specified option
+        # and ignore everything after.
+        pronoun_loc = doc["context_de"].index("_")
+        return doc["context_de"][:pronoun_loc] + option
```

Comment:
We don't need a whitespace as a prefix for the context, because the prompt starts with the context and the whitespace is only added when the context and target are added for the prompting